test/podFailover/workload/cmd/main.go

```diff
@@ -0,0 +1,347 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"log"
+	"net/http"
+	"net/url"
+	"os"
+	"path/filepath"
+	"strconv"
+	"time"
+
+	"github.com/Azure/azure-amqp-common-go/v3/aad"
+	eventhub "github.com/Azure/azure-event-hubs-go/v3"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/klog/v2"
+	podfailure "sigs.k8s.io/azuredisk-csi-driver/test/podFailover/api"
+)
+
+var mountPath = flag.String("mount-path", "/", "The path of the file where timestamps will be logged")
+var metricsEndpoint = flag.String("metrics-endpoint", "", "The endpoint where prometheus metrics should be published")
+var authEnabled = flag.Bool("auth-enabled", false, "Specify whether request to metrics endpoint requires authentication or not")
+var runID = flag.String("run-id", "", "The run id of the test")
+var storageProvisioner = flag.String("storage-provisioner", "", "The underlying storage provisioner of test")
+var workloadType = flag.String("workload-type", "default-workload-type", "The type of test being run (1pod1pvc, 3pod3pvc, etc)")
+var driverVersion = flag.String("driver-version", "", "The version of csi-driver")
+var namespace = flag.String("namespace", "pod-failover", "The namespace resources are created in")
+var kubeconfig = flag.String("kube-config", "", "kube config path for cluster")
+
+type DowntimeMetric struct {
+	TimeStamp     string `json:"timeStamp"`
+	RunID         string `json:"runID"`
+	StorageType   string `json:"storageType"`
+	DriverVersion string `json:"driverVersion"`
+	ImageVersion  string `json:"imageVersion"`
+	FailureType   string `json:"failureType"`
+	WorkloadType  string `json:"workloadType"`
+	PodDowntime   int64  `json:"downtime"`
+}
+
+func main() {
+	flag.Parse()
+	var config *rest.Config
+	var err error
+	if *kubeconfig != "" {
+		config, _ = clientcmd.BuildConfigFromFlags("", *kubeconfig)
+	} else {
+		config, err = rest.InClusterConfig()
+		if err != nil {
+			kubeConfigPath := os.Getenv("KUBECONFIG")
```

Comment:
Like `--kube-config`, the `KUBECONFIG` environment variable should override the in-cluster config. We can address this in a separate PR.